lint/indexOnlyLoop_checker.go

```diff
@@ -0,0 +1,81 @@
+package lint
+
+import (
+	"fmt"
+	"go/ast"
+)
+
+//! Detects when range loop is iterating over container which has pointer-type elements (that is, any *T or reference-like type as map or chan) using only key variable (no value) and container[key] occurs more than once inside loop body
```

Comment:
Can we make summary shorter?
Maybe something like this is concise enough:
`Detects for loops that can benefit from rewrite to range loop.`

Reply:
fixed